### PR TITLE
fix(stream agg): avoid lossy row count cast to help locate possible bug

### LINKED ISSUE #11216: Detect stream agg's silent error
## when row count minus to negative number

https://github.com/risingwavelabs/risingwave/blob/c6b490f5c41fef401b74577d21a9eed66b496635/src/stream/src/executor/aggregation/agg_group.rs#L242-L245

https://github.com/risingwavelabs/risingwave/blob/main/src/stream/src/executor/aggregation/agg_group.rs#L338-L342

> Casting between two integers of the same size (e.g. i32 -> u32) is a no-op (Rust uses 2's complement for negative values of fixed integers)

https://doc.rust-lang.org/reference/expressions/operator-expr.html#numeric-cast
And it is very important to `infer_change_type`. We should do a strict check like other operators with "sanity check" or handle the negative situation.

### PR DESCRIPTION
**Pull request by stdrc**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

- Avoid row count cast from `i64` to `usize` directly, fixes #11216. This also fixes over window function call like `lag(x, -1)` and `lead(x, -1)`;
- Add some tracing log to help locate bug.

## Checklist

- [x] I have written necessary rustdoc comments
- ~~[ ] I have added necessary unit tests and integration tests~~
- ~~[ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).~~
- ~~[ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).~~
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)
- ~~[ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)~~

- ~~[ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details]~~(https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- ~~[ ] My PR contains user-facing changes.~~

<details><summary>Click here for Documentation</summary>

### Types of user-facing changes

Please keep the types that apply to your changes, and remove the others.

- Installation and deployment
- Connector (sources & sinks)
- SQL commands, functions, and operators
- RisingWave cluster configuration changes
- Other (please specify in the release note below)

### Release note

</details>

### CODE REVIEW

File: src/stream/src/executor/aggregation/agg_group.rs [hunk 1]
```diff
@@ -237,16 +238,41 @@ impl<S: StateStore, Strtg: Strategy> AggGroup<S, Strtg> {
...
+            self.states[self.row_count_index],
+            AggState::Value(ref state) => state
+        );
+        let row_count = row_count_state
+            .get_output()
+            .expect("row count should never output NULL")
+            .into_int64();
+        Ok(row_count
+            .try_into()
+            .map_err(|_| anyhow!("row count should be non-negative, but got {}", row_count))?)
```
Comment (BugenZhao) [inline src/stream/src/executor/aggregation/agg_group.rs:273]: I guess it's okay to panic here directly.

Comment (stdrc) [reply src/stream/src/executor/aggregation/agg_group.rs:273]: By returning an error we can print more context info at the caller frame

Comment (xxchan) [reply src/stream/src/executor/aggregation/agg_group.rs:273]: Why panic can’t? What’s the difference? T👀

Comment (stdrc) [reply src/stream/src/executor/aggregation/agg_group.rs:273]: Returning an error to `AggGroup::apply_chunk` can enable us to log the current chunk processing?

Comment (stdrc) [reply src/stream/src/executor/aggregation/agg_group.rs:273]: But looks like logging the chunk won't help too much🤣 I don't know🥵

Comment (stdrc) [reply src/stream/src/executor/aggregation/agg_group.rs]: Changed.

**Review by BugenZhao — APPROVED** [review]

**Review by st1page — APPROVED** [review]
LGTM

**Pull request merged**